platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -34,6 +34,16 @@ const DEFAULT_STATE = {
 
 export const ViewportGridContext = createContext(DEFAULT_STATE);
 
+const isReuseableViewport = (oldViewport, newViewport) => {
```

Comment:
This code isn't strictly necessary, but it does reuse viewports when they are the same size/display sets, and are immediately redisplayed, and otherwise gets a new viewport id every time.